packages/generator/src/edmx-to-vdm/v4/operation.ts

```diff
@@ -20,95 +29,280 @@ const logger = createLogger({
 
 const extractResponse = (response: string) => `${response}.value`;
 
-interface JoinedOperationData {
-  operationImport: EdmxOperationImport;
-  operation: EdmxOperation;
+function splitMissingOperation(
+  operationImports: EdmxOperationImport[],
+  operations: EdmxOperation[]
+): [EdmxOperationImport[], EdmxJoinedOperation[]] {
+  return operationImports.reduce<
+    [EdmxOperationImport[], EdmxJoinedOperation[]]
+  >(
+    ([withoutOperation, withOperation], curr) => {
+      const operation = findOperationByImportName(
+        operations,
+        curr.operationName
+      );
+      if (operation) {
+        withOperation.push({
+          Name: curr.Name,
+          operationName: curr.operationName,
+          Parameter: operation.Parameter,
+          ReturnType: operation.ReturnType,
+          IsBound: operation.IsBound.toLowerCase() === 'true',
+          operationType: curr.operationType
+        });
+      } else {
+        withoutOperation.push(curr);
+      }
+      return [withoutOperation, withOperation];
+    },
+    [[], []]
+  );
+}
+
+function extractEntitySetName(type: string): string {
+  const components = type.split('.');
+  return components.pop()!;
+}
+
+function splitMissingParameter( // fixme name
```

Comment:
[req] here the name it is more than a split. Transformation of data.

Reply:
not sure what a good name would be.

Reply:
Perhaps you can use the `partition`of the array inside the method and add the setting of the parameter in a separate method with a good name.